dbt/adapters/databricks/impl.py

```diff
@@ -39,13 +39,15 @@ class DatabricksAdapter(SparkAdapter):
 
     def list_schemas(self, database: Optional[str]) -> List[str]:
         """Get a list of existing schemas in database."""
-        results = self.connections.list_schemas(database=database)
+        if database is not None:
+            results = self.connections.list_schemas(database=database)
+        else:
+            results = self.execute_macro(LIST_SCHEMAS_MACRO_NAME, kwargs={"database": database})
         return [row[0] for row in results]
 
     def check_schema_exists(self, database: Optional[str], schema: str) -> bool:
         """Check if a schema exists."""
```

Comment:
ditto

Reply:
Maybe we don't need here as it call `list_schemas` now?

Reply:
Sounds good.